tests/main/ubuntu-core-custom-device-reg-extras/task.yaml

```diff
@@ -0,0 +1,74 @@
+summary: |
+   Test that device initialisation and registration can be customized
+   with the prepare-device gadget hook and this can set request headers,
+   a proposed serial and the body of the serial assertion
+systems: [ubuntu-core-16-64]
+prepare: |
+    if [ "$TRUST_TEST_KEYS" = "false" ]; then
+        echo "This test needs test keys to be trusted"
+        exit
+    fi
+    . $TESTSLIB/systemd.sh
+    systemctl stop snapd.service snapd.socket
+    rm -rf /var/lib/snapd/assertions/*
+    rm -rf /var/lib/snapd/device
+    rm -rf /var/lib/snapd/state.json
+    unsquashfs /var/lib/snapd/snaps/pc_*.snap
+    mkdir -p squashfs-root/meta/hooks
+    cp prepare-device squashfs-root/meta/hooks
+    mksquashfs squashfs-root pc_x1.snap -comp xz
```

Comment:
You could use `snapbuild` here but in practise it does not make a difference currently.